sydent/db/sqlitedb.py

```diff
@@ -132,13 +132,30 @@ def _upgradeSchema(self):
             self.db.commit()
             logger.info("v0 -> v1 schema migration complete")
             self._setSchemaVersion(1)
+
         if curVer < 2:
             cur = self.db.cursor()
             cur.execute("CREATE INDEX threepid_validation_sessions_mtime ON threepid_validation_sessions(mtime)")
             self.db.commit()
             logger.info("v1 -> v2 schema migration complete")
             self._setSchemaVersion(2)
 
+        if curVer < 3:
+            cur = self.db.cursor()
+            cur.execute("ALTER TABLE local_threepid_associations ADD COLUMN lookup_hash VARCHAR(256)")
+            cur.execute("CREATE INDEX IF NOT EXISTS lookup_hash_medium on local_threepid_associations (lookup_hash, medium)")
+
+            cur.execute("ALTER TABLE global_threepid_associations ADD COLUMN lookup_hash VARCHAR(256)")
+            cur.execute("CREATE INDEX IF NOT EXISTS lookup_hash_medium on global_threepid_associations (lookup_hash, medium)")
+
+            cur.execute(
+                "CREATE TABLE IF NOT EXISTS hashing_metadata "
+                "(lookup_pepper varchar(256))"
+            )
+            self.db.commit()
+            logger.info("v2 -> v3 schema migration complete")
+            self._setSchemaVersion(3)
```

Comment:
Don't we need to go back and hash the existing stuff?

Reply:
That will be done on first run. Sydent checks to see if a pepper is configured in the database, and if not, generate one and rehash all 3PIDs:

https://github.com/matrix-org/sydent/blob/424270b144a2bc84bba4fb673da1c9b5180c9c21/sydent/sydent.py#L186-L194

Reply:
Hmm, OK. Though please add a comment there to say that it must be run *before* we start serving requests.